### PR TITLE
fix rootlessport flake

### LINKED ISSUE #11248: network connect/disconnect: new flake
Followup to #11091: looks like there's still something flaking:
```
[+1340s] not ok 241 podman network connect/disconnect with port forwarding
         # (from function `is' in file test/system/helpers.bash, line 474,
         #  in test file test/system/500-networking.bats, line 445)
         #   `is "$output" "$random_1" "curl 127.0.0.1:/index.txt should work again"' failed with status 56
         # $ podman rm --all --force
         # $ podman ps --all --external --format {{.ID}} {{.Names}}
         # $ podman images --all --format {{.Repository}}:{{.Tag}} {{.ID}}
         # quay.io/libpod/testimage:20210610 9f9ec7f2fdef
         # $ podman network create testnet-FA2Qkj7JP6
         # /home/some24215dude/.config/cni/net.d/testnet-FA2Qkj7JP6.conflist
         # $ podman network create testnet2-HVIhBV5Cvx
         # /home/some24215dude/.config/cni/net.d/testnet2-HVIhBV5Cvx.conflist
         # $ podman run -d --network testnet-FA2Qkj7JP6 quay.io/libpod/testimage:20210610 top
         # 0b007115b381d4c40e11b6948839dbf4d572fa899d5ec5dc6096a566dcc0b89d
         # $ podman run -d -p 12345:80 --network testnet-FA2Qkj7JP6 -v /tmp/podman_bats.nacKs4/hello.txt:/var/www/index.txt:Z -w /var/www quay.io/libpod/testimage:20210610 /bin/busybox-extras httpd -f -p 80
         # 5f10d636830a5d8aea5648a735386ff6471639fcdf7ef19bc859ec2bce349df9
         # $ podman inspect 5f10d636830a5d8aea5648a735386ff6471639fcdf7ef19bc859ec2bce349df9 --format {{(index .NetworkSettings.Networks "testnet-FA2Qkj7JP6").IPAddress}}
         # 10.89.0.3
         # $ podman inspect 5f10d636830a5d8aea5648a735386ff6471639fcdf7ef19bc859ec2bce349df9 --format {{(index .NetworkSettings.Networks "testnet-FA2Qkj7JP6").MacAddress}}
         # b6:f0:5d:b8:8f:81
         # $ podman network disconnect testnet-FA2Qkj7JP6 5f10d636830a5d8aea5648a735386ff6471639fcdf7ef19bc859ec2bce349df9
         # $ podman network connect testnet-FA2Qkj7JP6 5f10d636830a5d8aea5648a735386ff6471639fcdf7ef19bc859ec2bce349df9
         # time="2021-08-11T15:40:40-05:00" level=warning msg="Could not reload rootless port mappings, port forwarding may no longer work correctly: dial unix /run/user/6817/libpod/tmp/rp/5f10d636830a5d8aea5648a735386ff6471639fcdf7ef19bc859ec2bce349df9: connect: connection refused"
         # #/vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
         # #|     FAIL: curl 127.0.0.1:/index.txt should work again
         # #| expected: 'V2nBRSTMbI5tU89XO1xo5by7lYEdOT'
         # #|   actual: ''
         # #\^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
### sys: podman network connect/disconnect with port forwarding
* fedora-33 : sys podman fedora-33 rootless host
  * PR #11193
    * [08-11 16:43](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6156511840829440/html/sys-podman-fedora-33-rootless-host.log.html)
* fedora-34 : sys podman fedora-34 rootless host
  * PR #10589
    * [08-09 16:39](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4864496993828864/html/sys-podman-fedora-34-rootless-host.log.html)
* ubuntu-2010 : sys podman ubuntu-2010 rootless host
  * PR #11240
    * [08-17 07:37](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6539432569339904/html/sys-podman-ubuntu-2010-rootless-host.log.html)
  * PR #11125
    * [08-04 05:02](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4748189765992448/html/sys-podman-ubuntu-2010-rootless-host.log.html)

### PR DESCRIPTION
**Pull request by Luap99**
When the rootlessport process is started the stdout/stderr are attached
to the podman process. However once everything is setup podman exits and
when the rootlessport process tries to write to stdout it will fail with
SIGPIPE. The code handles this signal and puts /dev/null to stdout and
stderr but this is not robust. I do not understand the exact cause but
sometimes the process is still killed by SIGPIPE. Either go lost the
signal or the process got already killed before the goroutine could
handle it.

Instead of handling SIGPIPE just set /dev/null to stdout and stderr
before podman exits. With this there should be no race and no way to
run into SIGPIPE errors.

[NO TESTS NEEDED]

Fixes #11248

### CODE REVIEW

**Comment by Luap99** [comment]
@edsantiago I have run this patch for over one hour in the test vm without failure so I hope this no longer flakes.

**Comment by edsantiago** [comment]
Thank you!

**Comment by mheon** [comment]
LGTM

File: pkg/rootlessport/rootlessport_linux.go [hunk 1]
```diff
@@ -251,8 +226,16 @@ outer:
 		go serve(socket, driver)
 	}
 
-	// write and close ReadyFD (convention is same as slirp4netns --ready-fd)
 	logrus.Info("ready")
+
+	// https://github.com/containers/podman/issues/11248
+	// Copy /dev/null to stdout and stderr to prevent SIGPIPE errors
+	if f, err := os.OpenFile("/dev/null", os.O_WRONLY, 0755); err == nil {
```
Comment (rhatdan) [inline pkg/rootlessport/rootlessport_linux.go:233]: Nit: Could have used os.DevNull, 

LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request merged**